.github/workflows/test-changed-firestore.yml

```diff
@@ -28,27 +34,112 @@ jobs:
           cp config/ci.config.json config/project.json
           yarn
       - name: build
-        run: yarn build:changed firestore
-      - name: Run tests if firestore or its dependencies has changed
-        run: yarn test:changed firestore
+        id: build
+        # TODO(wuandy): Separate yarn and egrep into steps, so build failure
+        # is captured by github actions.
+        run: yarn build:changed firestore | egrep "Skipping all"
+        # Continue when "Skipping all" is not found
+        continue-on-error: true
+      - name: set output
+        # This means "Skipping all" was not found
+        if: steps.build.outcome != 'success'
+        id: set-output
+        run: echo "CHANGED=true" >> "$GITHUB_OUTPUT";
+      - name: Archive build
+        if: ${{ !cancelled() && steps.build.outcome != 'success' }}
+        run: |
+          tar -cf build.tar --exclude="\.git" .
+          gzip build.tar
+      - name: Upload build archive
+        if: ${{ !cancelled() && steps.build.outcome != 'success' }}
+        uses: actions/upload-artifact@v3
+        with:
+          name: build.tar.gz
+          path: build.tar.gz
+          retention-days: ${{ env.artifactRetentionDays }}
+
+  compat-test-chrome:
```

Comment:
Should we also add a `compat-test-firefox`? It looks like the old code ran the compat tests on firefox as well.

Reply:
Just a reminder to make sure to add `compat-test-firefox` to this repo's "required" checks when you add it.